### PR TITLE
Step By Step Instructions for Building and Running the Solution

### PR DESCRIPTION
**Pull request by JohnPWSharp**
Work in progress
connects to #37

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@JohnPWSharp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: docs/building-and-deploying-the-sample-solution.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+- Rebuild the solution. This action downloads the various NuGet packages referenced by the solution and installs the appropriate assemblies.
+
+> **Note:** The rebuild process should report one error: *Before building this project, please copy mysettings-template.config to mysettings.config ...* You will do this in a later step. If rebuild process displays any other errors then clean the solution and rebuild it again.
+
+### Creating the Required Azure Assets and Services
+
+- Using the Azure web portal, manually create the following items:
+
+	- [A Service Bus namespace](https://msdn.microsoft.com/library/azure/hh690931.aspx). The namespace should be created in the Central US region. Set the namespace type to Notification Hub.
```
Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: This is included as an step in the scripts.

Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: This is included as an step in the scripts.

Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: This is included as an step in the scripts.

File: docs/building-and-deploying-the-sample-solution.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+	- [A storage account](https://azure.microsoft.com/documentation/articles/storage-create-storage-account/). Set the location/affinity group to Central US.
```
Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: Storage Account is created as part of the provisioning scripts.

**Comment by bennage** [comment]
Regarding your questions:
> 1. What do the 5 scenarios actually demonstrate (I can guess 1, 2, and 4, but 3 and 5 are not clear)?

We're going to remove the extraneous scenarios per #69 

> 2. What do the Unit tests actually test?

Basic behaviors of the simulator. These were ported over from data-pipeline. Some where added because we had some design bugs, and the tests were used to debug and prevent regression. I don't think we need to say much in the docs. I'd be more in favor of adding explanatory comments in the test project. Thoughts?

> 3. Where does the output from the Hive query go (the hivequery1 script just displays the Hive job number and indicates that it has been submitted successfully)

I thought that it was output to a file (or stdout). @hanzzhang can you verify?

**Comment by bennage** [comment]
@hanzzhang just verified that it should be stdout.

**Comment by JohnPWSharp** [comment]
I have tried several times (using the provisioning script for HDInsight provided by Hernan), and I still get no output apart from:

Submitting Hive query.. Started Hive query with jobDetails Id : job_1433168880704_0001 Hive query completed Successfully.

File: docs/building-and-deploying-the-sample-solution.md (hunk 1 above)
Comment (JohnPWSharp) [inline docs/building-and-deploying-the-sample-solution.md]: Updated in the latest version of this doc

File: docs/building-and-deploying-the-sample-solution.md (hunk 2 above)
Comment (JohnPWSharp) [inline docs/building-and-deploying-the-sample-solution.md]: Updated in the latest version of this doc

**Comment by lbrader** [comment]
I'm getting the same out put as @JohnPWSharp .

File: docs/building-and-deploying-the-sample-solution.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+- On the Stream Analytics page, verify that a Stream Analytics job has been created using the name that you specified when you ran the provisioning script (the name that you entered in response to the *StreamAnalyticsJobName* prompt).
+
+- Click the job, and then in the menu bar click Configure
+
+- On the Configuration page, verify that the storage account is set to the value that you specified when you ran the provisioning script (the name that you entered in response to the *StorageAccountName* prompt).
+
+- In the menu bar click Inputs.
+
+- On the Inputs page, verify that an input named input01 has been created and that the type of the input is Event Hub.
```
Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: Input02 - ReferenceData - BlobStorage is also being created. Should we add that?

File: docs/building-and-deploying-the-sample-solution.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+- In the Serialization section, verify that the event serialization format is set to JSON.
+
+- Return to the job page, and then in the menu bar click Query
+
+- On the Query page, verify that the following query is specified:
+	`SELECT * FROM input01`
+
+- Return to the job page, and then in the menu bar click Outputs.
+
+- On the Outputs page, verify that an output named output01 has been created and that the output type is set to Blob Storage
```
Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: An Output02 - SqlServer is also created by the provisioning scripts.

File: docs/building-and-deploying-the-sample-solution.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+The following sections describe how to test and run the simulator.
+
+### Running the Unit Tests
+
+The UnitTests project ... (*What do these tests demonstrate? - TBD*)
+
+To run the unit tests, on the Visual Studio menu bar click Test, click Run, and then click All Tests.
+
+All tests should succeed without errors.
```
Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: Before running the tests you have to change App.config to add an storage account and account key, if you don´t do this many tests will fail.

I didn´t find it in the portal, but to get the storage account key you can execute 

GetAzureStorageKey -StorageAccountName <<StorageAccountName>>

from an Azure Powershell console.

File: docs/building-and-deploying-the-sample-solution.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+- On the menu, select option 1 (*Run NoErrorsExpected*). The simulator will generate events and echo their contents to the screen. The simulator should not report any exceptions.
+
+- Allow the simulator to run for a couple of minutes and then press q.
+
+- Press Enter to return to the menu.
+
+- Repeat the previous three steps for options 2 to 5. As before, the simulator will generate and echo events and no exceptions should occur.
+
+- On the menu, select option 6 to quit the simulator.
```
Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: Option to exit is [3] not [6]

File: docs/building-and-deploying-the-sample-solution.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+> **Note:** Windows Powershell must be configured to run scripts. You can do this by running the following PowerShell command before continuing:
+> 
+> `set-executionpolicy unrestricted`
+
+- Move to the folder containing the source code for the solution scripts, and then move to the Validation\HDInsight subfolder.
+
+- Run the following command:
+
+	`.\hivequery.ps1`
```
Comment (hveiras) [inline docs/building-and-deploying-the-sample-solution.md]: There are two different scripts now:

- hivequeryforcoldstorageeventprocessor.ps1
- hivequeryforstreamanalytics.ps1

**Comment by JohnPWSharp** [comment]
@francischeung @hanz @lbrader @bennage I am a little confused as to what the various components are now doing, and the purpose of the ColdStorage console host. I understand the flow of information from the devices through the event hub to stream analytics and thence to blob storage and the Hive query running in Hadoop. What I don't get is what the cold storage processor does, or how the SQL warm path fits in (what are the flows?).

IMHO It would be good to include a more detailed version of the diagram based on that shown in the "Introducing the Journey" doc to illustrate how the how/warm/cold paths are mapped by the example solution.

**Comment by bennage** [comment]
@francischeung and I were going over some of this yesterday. I think we have a way to communicate this. I'll write up a description for you to review.

**Comment by JohnPWSharp** [comment]
@bannage @francischeung The doc overview-of-architecture-and-components.md in this PR contains a first pass at a summary of the elements that comprise the sample solution and the flows for the hot, warm, and cold paths as they currently stand.

**Comment by bennage** [comment]
We identified some content that needs to be revised in this document, but we will handle that in a future pull request.

**Pull request merged**